docker-compose.yaml

```diff
@@ -0,0 +1,53 @@
+version: '3.1'
+
+services:
+
+  db:
+    image: mysql:5.7
+    restart: on-failure
+    environment:
+      MYSQL_DATABASE: testdb
+      MYSQL_USER: wordpress
+      MYSQL_PASSWORD: wordpress
+      MYSQL_RANDOM_ROOT_PASSWORD: '1'
+    volumes:
+      - db:/var/lib/mysql
+
+  wordpress-woocomerce-dev:
+    depends_on:
+      - db
+    build:
+      context: .
+      dockerfile: Dockerfile
+    ports:
+      - 8084:80
+    restart: on-failure
+    environment:
+      WORDPRESS_DB_HOST: db
+      WORDPRESS_DB_NAME: testdb
+      WORDPRESS_DB_USER: wordpress
+      WORDPRESS_DB_PASSWORD: wordpress
+      WORDPRESS_TABLE_PREFIX: "wp_"
+      WORDPRESS_DEBUG: 1
+    volumes:
+      - wordpress:/var/www/html
+
+  wordpress-cli:
+    depends_on:
+      - db
+      - wordpress-woocomerce-dev
+    image: wordpress:cli
+    restart: on-failure
+    user: xfs
+    command: >
+      /bin/sh -c '
+      wp core install --url=http://localhost:8084 --title="WooCommerce Core E2E Test Suite" --admin_user=admin --admin_password=password --admin_email=admin@woocommercecoree2etestsuite.com --path=/var/www/html --skip-email;
+      wp plugin activate woocommerce;
```

Comment:
I am wondering since the source is coming from github it might be missing the build dependencies, so before activating we might need to run `composer install --no-dev` and `npm install`

Reply:
That's right, it does miss that. Thanks for catching it! I am not sure it needs to be done within a Docker container. 

When running tests locally, we can run it before we run the tests. I updated the instructions on how to test this PR locally with these 2 commands. 

When it comes to Travis, I see that we run these commands in the `before_script` section of `travis.yml`:

`composer install --no-dev`:

https://github.com/woocommerce/woocommerce/blob/master/.travis.yml#L66

`npm install` is currently being run here within `install_e2e_site()` which we no longer need:

https://github.com/woocommerce/woocommerce/blob/master/tests/bin/install.sh#L222

I added it to `run-e2e-CI.sh` instead of before running tests in the new Travis implementation.